include/dlaf/gpu/blas/error.h

```diff
@@ -18,65 +18,26 @@
 #include "dlaf/common/source_location.h"
 #include "dlaf/gpu/blas/api.h"
 #include "dlaf/gpu/cublas/error.h"
-#include "dlaf/gpu/hipblas/error.h"
+#include "dlaf/gpu/rocblas/error.h"
 
 namespace dlaf::gpublas::internal {
 
 #ifdef DLAF_WITH_GPU
 
 inline void checkError(gpublasStatus_t st,
                        const dlaf::common::internal::source_location& info) noexcept {
+  if (st != GPUBLAS_STATUS_SUCCESS) {
 #ifdef DLAF_WITH_CUDA
-  if (st != CUBLAS_STATUS_SUCCESS) {
     std::cout << "[CUBLAS ERROR] " << info << std::endl
               << cublas::internal::getErrorString(st) << std::endl;
-    std::terminate();
-  }
 #elif defined(DLAF_WITH_HIP)
-  if (st != HIPBLAS_STATUS_SUCCESS) {
-    std::cout << "[HIPBLAS ERROR] " << info << std::endl
-              << hipblas::internal::getErrorString(st) << std::endl;
-    std::terminate();
-  }
+    std::cout << "[ROCBLAS ERROR] " << info << std::endl
+              << rocblas::internal::getErrorString(st) << std::endl;
 #endif
-}
-
-#ifdef DLAF_WITH_HIP
-// Lifted from hipBLAS/library/src/hcc_detail/hipblas.cpp, with missing
-// enumeration values added. This should be removed when hipBLAS is no longer
-// used.
-inline hipblasStatus_t rocBLASStatusToHIPStatus(rocblas_status error) {
-  switch (error) {
-    case rocblas_status_continue:
-    case rocblas_status_size_unchanged:
-    case rocblas_status_size_increased:
-    case rocblas_status_success:
```

Comment:
Remember that these all need to be considered success with rocblas. They are returned when getting the workspace size.

Reply:
@msimberg I'm wondering if separating the two thing would be better:
- when executing `rocblas_status_success` is the expected return code and is checked with `DLAF_GPUBLAS_CHECK_ERROR`
- when checking the rocblas workspace the expected return code are `rocblas_status_continue, rocblas_status_size_increased, rocblas_status_size_unchanged` which should be checked with a different method. (A possible name can be`DLAF_ROCBLAS_WORKSPACE_CHECK_ERROR`).

Reply:
Yeah, that sounds reasonable.